collect_app/src/test/java/org/odk/collect/android/widgets/VideoWidgetTest.java

```diff
@@ -0,0 +1,92 @@
+package org.odk.collect.android.widgets;
+
+
+import android.content.Context;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+
+import net.bytebuddy.utility.RandomString;
+
+import org.javarosa.core.model.data.StringData;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.odk.collect.android.BuildConfig;
+import org.odk.collect.android.utilities.FileUtil;
+import org.odk.collect.android.utilities.MediaUtil;
+import org.odk.collect.android.widgets.base.BinaryNameWidgetTest;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import java.io.File;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author James Knight
+ */
+@Config(constants = BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+public class VideoWidgetTest extends BinaryNameWidgetTest<VideoWidget> {
+
+    @Mock
+    Uri uri;
+
+    @Mock
+    MediaUtil mediaUtil;
+
+    @Mock
+    FileUtil fileUtil;
+
+    private String destinationName = null;
+
+    public VideoWidgetTest() {
+        super(VideoWidget.class);
+    }
+
+    @NonNull
+    @Override
+    public VideoWidget createWidget() {
+        VideoWidget videoWidget = new VideoWidget(RuntimeEnvironment.application, formEntryPrompt);
+
+        videoWidget.setMediaUtil(mediaUtil);
+        videoWidget.setFileUtil(fileUtil);
+
+        return videoWidget;
+    }
+
+    @NonNull
+    @Override
+    public StringData getNextAnswer() {
+        return new StringData(destinationName);
+    }
+
+    @Override
+    public Object createBinaryData(StringData answerData) {
+        return uri;
+    }
+
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        when(formEntryPrompt.isReadOnly()).thenReturn(false);
+
+        when(mediaUtil.getPathFromUri(any(Context.class), any(Uri.class), any(String.class)))
+                .thenReturn(String.format("%s.mp3", RandomString.make()));
```

Comment:
It doesn't matter for the test but strictly speaking this should probably be mp4.

Reply:
Agreed, this was a copy/paste from the near identical AudioWidgetTest code.